model/device.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2021 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package model
+
+// Device represents a device and its available integrations
+type Device struct {
+	// ID is the device ID
+	ID string `json:"id" bson:"_id"`
+	// Integrations contains the list of integrations for this device
+	Integrations []DeviceIntegration `json:"integrations" bson:"integrations"`
```

Comment:
We will probably at some point need to be able to query devices that belong to a given integration, in that case it we can do this efficiently if we have a multi key index on a field holding the list of integration IDs.
If we need additional metadata stored within the device document, I would propose adding another embedded document which maps integration IDs to meta-data.
```suggestion
	// Integrations contains the list of integrations for this device
	IntegrationIDs []uuid.UUID `json:"?" bson:"integration_ids"`
	
	Integrations map[string]MetaData `json:"integrations,omitempty" bson:"integrations,omitempty"`
```

Reply:
We don't need the meta data, let's go with the IntegrationIDs only.